polymer/src/amp-my-embed.html

```diff
@@ -0,0 +1,155 @@
+Code distributed by Google as part of the polymer project is also
+subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+-->
+
+<link rel="import" href="../node_modules/@npm-polymer/polymer/polymer-element.html">
+<link rel="import" href="../node_modules/@npm-polymer/iron-flex-layout/iron-flex-layout.html">
+<script async src="https://cdn.ampproject.org/shadow-v0.js"></script>
+
+<dom-module id="amp-my-embed">
+
+  <template>
+
+    <style>
+
+      :host {
+        @apply --layout-vertical;
+        @apply --layout-center-justified;
+        text-align: center;
+        margin: 0 48px;
+      }
+
+      .title {
+        color: var(--app-primary-color);
+        font-weight: bold;
+      }
+
+      .description {
+        color: var(--app-secondary-color);
+      }
+
+      @media (max-width: 767px) {
+        :host {
+          margin: 0 12px;
+        }
+      }
+
+    </style>
+
+    <span id="ampContent"></span>
+
+  </template>
+
+  <script>
+
+    class AmpMyEmbed extends Polymer.Element {
+
+      static get is() { return 'amp-my-embed'; }
+
+      static get properties() { return {
+
+        url: {
+          type: String,
+          value: window.location.origin
+        },
+        fullUrl: {
+          type: String,
+          computed: '_computeFullUrl(url, itemUrl)',
+          observer: 'urlChanged'
+        },
+        itemUrl: {
+          type: String,
+          notify: true
+        },
+        content: {
+          type: String,
+          notify: true
+        }
+      };
+    }
+
+      _computeFullUrl(url, itemUrl) {
+        return [url, itemUrl, "embed"].join('')
+      }
+
+      urlChanged() {
+        this.fetchAndAttachAmpDoc_(this.fullUrl);
+      }
+
+      constructor() {
+        super();
+        // 'offline' is set to true if and when the document fetch fails.
+        this.state = {'offline': false};
+        this.ampReadyPromise_ = new Promise(resolve => {
+          (window.AMP = window.AMP || []).push(resolve);
+        });
+
+        this.xhr_ = null;
+        this._amp = null;
+        this._host = null;
+      }
+
+      ready() {
+        super.ready();
+        this.$.ampContent.textContent = this._amp;
+      }
+
+  /**
+   * Fetches the AMP document at `url` and attaches it as a shadow root.
+   * @private
+   * @param {string} url
+   */
+  fetchAndAttachAmpDoc_(url) {
+    var myHeaders = new Headers();
+    myHeaders.append("Accept'", "text/*");
+    myHeaders.append('Content-Type', 'document');
+
+    var myInit = { method: 'GET',
+                   headers: myHeaders};
+
+    fetch(url, myInit).then(function(response) {
+      return response.text();
+    }).then(doc => {
+      return this.ampReadyPromise_.then(amp => {
+
+        var parser = new DOMParser(),
+        newDoc = parser.parseFromString( doc, 'text/html' );
+
+        // Replace the old shadow root with a new div element.
+        const oldShadowRoot = this._host;
```

Comment:
you're mixing const and var. Either only use const & let or use var everywhere.

Reply:
Using var